easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -589,10 +589,19 @@ def _parse_dependency(self, dep, hidden=False):
             'external_module_metadata': {},
         }
         if isinstance(dep, dict):
-            dependency.update(dep)
             # make sure 'dummy' key is handled appropriately
             if 'dummy' in dep and not 'toolchain' in dep:
                 dependency['toolchain'] = dep['dummy']
+            if 'external' in dep and dep['external']:
+                dependency['external_module'] = True
+                dependency['short_mod_name'] = dep['name']
+                dependency['full_mod_name'] = dep['name']
+                if dep['name'] in self.external_modules_metadata:
+                        dependency['external_module_metadata'].update(self.external_modules_metadata[dep['name']])
+                        self.log.info("Updated dependency info with available metadata for external module %s: %s",
+                                      dep['name'], dependency['external_module_metadata'])
```

Comment:
this is almost a copy-paste from stuff below, so this should be fleshed out in a separate function (one without side-effects), something like:

```python
if dep.get('external', False):
    dependency.update(self.handle_external_module_dep(dep['name']))
```

and then similar below with `dep[0]` as argument

the tricky bit is going to be the `dependency['external_module_metadata'].update` ?